manager/volume.go

```diff
@@ -765,21 +775,86 @@ func (m *VolumeManager) CancelExpansion(volumeName string) (v *longhorn.Volume,
 	return v, nil
 }
 
-func (m *VolumeManager) UpdateRecurringJobs(volumeName string, jobs []types.RecurringJob) (v *longhorn.Volume, err error) {
+func (m *VolumeManager) AddVolumeRecurringJob(volumeName string, name string, isGroup bool) (volumeRecurringJob map[string]*types.VolumeRecurringJob, err error) {
+	defer func() {
+		err = errors.Wrapf(err, "unable to update volume recurring jobs for %v", volumeName)
+	}()
+
+	v, err := m.ds.GetVolume(volumeName)
+	if err != nil {
+		return nil, err
+	}
+
+	key := ""
+	if isGroup {
+		key = types.GetRecurringJobLabelKey(types.LonghornLabelCronJobGroup, name)
+	} else {
+		key = types.GetRecurringJobLabelKey(types.LonghornLabelCronJob, name)
+	}
+	v.Labels[key] = types.LonghornLabelValueEnabled
```

Comment:
Check if it does not exist before addition.

Reply:
Fixed